From: "Jyri J. Virkki" <jyri@virkki.com>
Date: Thu, 23 Feb 2017 23:30:21 -0800
Subject: Fix shared library names on OS X build.

---
 Makefile | 31 ++++++++++++++++++++++---------
 1 file changed, 22 insertions(+), 9 deletions(-)

diff --git a/Makefile b/Makefile
index e02a9ea..d2b9200 100644
--- a/Makefile
+++ b/Makefile
@@ -20,8 +20,9 @@
 #   make clean          the usual
 #
 
-BLOOM_VERSION=1.4
-BLOOM_SONAME=libbloom.so.1
+BLOOM_VERSION_MAJOR=1
+BLOOM_VERSION_MINOR=4
+BLOOM_VERSION=$(BLOOM_VERSION_MAJOR).$(BLOOM_VERSION_MINOR)
 
 TOP := $(shell /bin/pwd)
 BUILD_OS := $(shell uname)
@@ -36,26 +37,38 @@ ifeq ($(MM),)
 MM=-m64
 endif
 
+#
+# Shared library names - these definitions work on most platforms but can
+# be overridden in the platform-specific sections below.
+#
+BLOOM_SONAME=libbloom.so.$(BLOOM_VERSION_MAJOR)
+SO_VERSIONED=libbloom.so.$(BLOOM_VERSION)
+LD_SONAME=-Wl,-soname,$(BLOOM_SONAME)
+SO=so
+
+
 ifeq ($(BUILD_OS),Linux)
 RPATH=-Wl,-rpath,$(BUILD)
-SO=so
 endif
 
 ifeq ($(BUILD_OS),SunOS)
 RPATH=-R$(BUILD)
-SO=so
 CC=gcc
 endif
 
 ifeq ($(BUILD_OS),OpenBSD)
 RPATH=-R$(BUILD)
-SO=so
 endif
 
 ifeq ($(BUILD_OS),Darwin)
-MAC=-install_name $(BUILD)/libbloom.dylib
+MAC=-install_name $(BUILD)/libbloom.dylib \
+	-compatibility_version $(BLOOM_VERSION_MAJOR) \
+	-current_version $(BLOOM_VERSION)
 RPATH=-Xlinker -rpath -Xlinker $(BUILD)
 SO=dylib
+BLOOM_SONAME=libbloom.$(BLOOM_VERSION_MAJOR).$(SO)
+SO_VERSIONED=libbloom.$(BLOOM_VERSION).$(SO)
+LD_SONAME=
 endif
 
 ifeq ($(DEBUG),1)
@@ -70,9 +83,9 @@ all: $(BUILD)/libbloom.$(SO) $(BUILD)/libbloom.a
 $(BUILD)/libbloom.$(SO): $(BUILD)/murmurhash2.o $(BUILD)/bloom.o
 	(cd $(BUILD) && \
 	    $(COM) $(LDFLAGS) bloom.o murmurhash2.o -shared $(LIB) $(MAC) \
-		-Wl,-soname,$(BLOOM_SONAME) -o libbloom.$(SO) && \
-		cp -p libbloom.$(SO) libbloom.$(SO).$(BLOOM_VERSION) && \
-		ln -s libbloom.$(SO).$(BLOOM_VERSION) $(BLOOM_SONAME))
+		$(LD_SONAME) -o libbloom.$(SO) && \
+		cp -p libbloom.$(SO) $(SO_VERSIONED) && \
+		ln -s $(SO_VERSIONED) $(BLOOM_SONAME))
 
 $(BUILD)/libbloom.a: $(BUILD)/murmurhash2.o $(BUILD)/bloom.o
 	(cd $(BUILD) && ar rcs libbloom.a bloom.o murmurhash2.o)
